lib/vrl/parser/src/template_string.rs

```diff
@@ -0,0 +1,76 @@
+use diagnostic::Span;
+use std::fmt;
+
+use crate::ast::{Expr, Ident, Literal::RawString, Node, Op, Opcode};
+
+#[derive(Clone, PartialEq, Eq, Ord, PartialOrd, Debug, Hash)]
+pub struct TemplateString(pub Vec<StringSegment>);
+
+impl TemplateString {
+    /// Rewrites the ast for the template string to be a series of string concatenations
+    pub fn rewrite(&self) -> Expr {
```

Comment:
I was going to suggest using a similar language as discussed [here](https://rustc-dev-guide.rust-lang.org/lowering.html):

```suggestion
    pub fn lower(&self) -> Expr {
```

But we're rewriting one AST node to another, so I guess that's not _exactly_ apples to apples.

Let's go with `rewrite_to_concatenated_strings` instead, to make it more clear what's going on at the call-site of this function.